patches/server/1005-PreSpawnerSearchEvent.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:53:44 +1000
+Subject: [PATCH] PreSpawnerSearchEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/level/BaseSpawner.java b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+index 633500aefd515df5dadda3802b94079f75a03fa0..408d360679e2acafa8fb711b92ec851f7d7d1648 100644
+--- a/src/main/java/net/minecraft/world/level/BaseSpawner.java
++++ b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+@@ -46,6 +46,7 @@ public abstract class BaseSpawner {
+     public int requiredPlayerRange = 16;
+     public int spawnRange = 4;
+     private int tickDelay = 0; // Paper
++    private int searchCooldownTicks = 0; // Paper
+ 
+     public BaseSpawner() {}
+ 
+@@ -57,6 +58,16 @@ public abstract class BaseSpawner {
+     public boolean isNearPlayer(Level world, BlockPos pos) {
+         return world.hasNearbyAlivePlayerThatAffectsSpawning((double) pos.getX() + 0.5D, (double) pos.getY() + 0.5D, (double) pos.getZ() + 0.5D, (double) this.requiredPlayerRange); // Paper - Affects Spawning API
+     }
++    // Paper Start - PreSpawnerSearchEvent (Convert ServerPlayers into BukkitPlayer)
++    public java.util.Collection<org.bukkit.entity.Player> convertServerPlayersToBukkitPlayers(java.util.List<net.minecraft.server.level.ServerPlayer> serverPlayers) {
```

Comment:
Can just be replaced by a `com.google.common.collect.Collections2.transform(serverPlayerList, ServerPlayer::getBukkitEntity)` in the code itself.